plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/controls/CustomFormEditor.java

```diff
@@ -301,17 +305,48 @@ public void widgetSelected(SelectionEvent e) {
             link.setLayoutData(new GridData(GridData.FILL_HORIZONTAL));
             return link;
         } else if (isTextPropertyType(propType)) {
-            if (property instanceof ObjectPropertyDescriptor && ((ObjectPropertyDescriptor) property).getLength() == PropertyLength.MULTILINE) {
+            if (property instanceof ObjectPropertyDescriptor && property.getLength() == PropertyLength.MULTILINE) {
                 Label label = UIUtils.createControlLabel(parent, propertyDisplayName);
                 label.setLayoutData(new GridData(GridData.VERTICAL_ALIGN_BEGINNING));
-                Text editor = new Text(parent, SWT.BORDER | SWT.MULTI | SWT.V_SCROLL | SWT.WRAP | (readOnly ? SWT.READ_ONLY : SWT.NONE));
-
+                Composite composite = UIUtils.createComposite(parent, 2);
+                Text editor = new Text(composite, SWT.MULTI | SWT.WRAP | SWT.BORDER | (readOnly ? SWT.READ_ONLY : SWT.NONE));
+                editor.setLayoutData(new GridData(GridData.FILL_BOTH));
                 editor.setText(objectValueToString(value));
-                GridData gd = new GridData(GridData.FILL_HORIZONTAL);
-                // Make multiline editor at least two lines height
-                gd.heightHint = (UIUtils.getTextHeight(editor) + editor.getBorderWidth()) * 2;
-                editor.setLayoutData(gd);
-                return editor;
+
+                int textHeight = UIUtils.getTextHeight(editor) * 3;
+                GridData gdEditor = new GridData(SWT.FILL, SWT.TOP, true, false);
+                gdEditor.heightHint = textHeight;
+                composite.setLayoutData(gdEditor);
+
+                Button editButton = new Button(composite, SWT.FLAT | SWT.PUSH);
+                editButton.setImage(DBeaverIcons.getImage(UIIcon.DOTS_BUTTON));
```

Comment:
Why don't we want to use a proper "pencil" icon, let's say?

Reply:
![image](https://github.com/user-attachments/assets/753432c9-1f76-4b8d-a5fb-d2a9ae9fb119)
I think it seems uglier than the dots icon. 
@serge-rider What do you think?

Reply:
Remove the border.

Reply:
@ShadelessFox Think this could be a little bit confusing because the user already could edit text, and that `pencil` said that you can also edit it. 
![image](https://github.com/user-attachments/assets/5c7bac69-4d8a-4668-84b7-d7dad40d84fc)

Reply:
For consistency reasons, `...` is nice. But its definition is too vague. It would be nicer to introduce a new icon as a replacement. Okay, let's keep dots for now.